### ISSUE TITLE
dump() arguments: stream vs dst

### ISSUE THREAD

**Issue by koct9i** [mk-fg/pretty-yaml#44]
There is tiny incompatibility with PyYAML:

PyYAML: dump(data, stream=None, ...) 
https://github.com/yaml/pyyaml/blob/main/lib/yaml/__init__.py#L248

pretty-yaml: dump(data, dst=None, ...)
https://github.com/mk-fg/pretty-yaml/blob/master/pyaml/__init__.py#L164

As a result:

```
>>> yaml.dump({}, stream=sys.stdout)
{}

>>> pyaml.dump({}, stream=sys.stdout)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/xxx/.local/lib/python3.10/site-packages/pyaml/__init__.py", line 181, in dump
    yaml.dump_all( data, buff, Dumper=Dumper,
TypeError: dump_all() got multiple values for argument 'stream'
```

**Comment by mk-fg** [comment]
Yeah, it's unfortunate, but that's the current API, and I don't think this is a good reason to break it.
Also, dst option works a bit differently from how stream does in pyyaml.

**Comment by mk-fg** [comment]
Oh, although guess stream= can be easily supported alongside dst=, which didn't occur to me at first.

**Comment by mk-fg** [comment]
Should be fixed in 23.9.0, by checking if either stream or dst is passed (but not both), and defaulting to the same old "return str" behavior.
Thanks for reporting!
